lib/Base/Tensor.cpp

```diff
@@ -282,6 +282,48 @@ static void transposeSelectImpl(const Handle<ElemTy> &src, Handle<ElemTy> &dest,
     transposeGenericImpl(src, dest, srcCoor, destCoor, shuffle);
   }
 }
+
+template <class ElemTy>
+static bool isTiledAlongAxisImpl(const Tensor *tensor, size_t axis,
+                                 size_t size) {
+  assert(axis < tensor->dims().size() && "Axis parameter invalid!");
+  assert(size <= tensor->dims()[axis] && "Size parameter invalid!");
+  assert(size >= 1 && "Size parameter invalid!");
+
+  // When the tile size matches the dimension size then we return true.
+  if (size == tensor->dims()[axis]) {
+    return true;
+  }
```

Comment:
```size == tensor->dims()[axis]``` is a very particular case where it makes sense to always return true because a tensor can be considered a tiled version of itself.
Indeed I did not think about the case ```tensor->dims()[axis] % size != 0``` and therefore I put an assert at the start of the function to verify that the tile size divides the dimension size for axis:
```
assert((tensor->dims()[axis] % size) == 0 && "Size parameter invalid!");
```

Reply:
Right, I was wondering about the "fractional tile" case. Is this an actual case you're expecting to be useful given the way it's used? If not (I don't think it is?) it feels a bit problematic, I'd probably add a check for this, or perhaps add an optional bool param for allowing it and default it to disallowed because AFAIK it's not commonly understood that way.

Reply:
I agree with you. What I did is this:
- I added an extra boolean flag `fractional` which by default is set to `false`. This flag enables the extra verification for fractional tiling but by default only `integer` tiling is verified.
- In the spirit of generalization, I also added a second flavor of the function for multiple axes and sizes. I also renamed the function to `isTiled()` and added extra unit tests.